csm/common/errors.py

```diff
@@ -152,11 +191,42 @@ class CsmGatewayTimeout(CsmError):
     a timely response from the upstream server.
     """
 
+    _desc = "Did not receive a timely response from the upstream server"
+
+    def __init__(self, desc=None, message_id=const.GATEWAY_TIMEOUT, message_args=None):
+        super(CsmGatewayTimeout, self).__init__(
+            CSM_GATEWAY_TIMEOUT, desc,
+            message_id, message_args)
 
 class CsmUnauthorizedError(CsmError):
 
     """This error represents HTTP 401 Unauthorized Error"""
 
+    _desc = "Invalid authentication credentials for the target resource."
+
+    def __init__(self, desc=None, message_id=const.UNAUTHORIZED_ERROR, message_args=None):
+        super(CsmUnauthorizedError, self).__init__(
+            CSM_UNAUTHORIZED_ERROR, desc,
+            message_id, message_args)
+
 class CsmServiceNotAvailable(CsmError):
 
     """This  error represents CSM service is Not Available."""
+
+    _desc = "CSM service is Not Available."
+
+    def __init__(self, desc=None, message_id=const.SERVICE_NOT_AVAILABLE, message_args=None):
+        super(CsmServiceNotAvailable, self).__init__(
+            CSM_SERVICE_NOT_AVAILABLE, desc,
+            message_id, message_args)
+
+class CsmRequestCancelled(CsmError):
+
+    """This  error represents CSM service request is cancelled."""
+
+    _desc = "CSM service request is cancelled."
```

Comment:
CSM service request is cancelled. --> Service request cancelled.

Reply:
updated